src/ui/public/kuery/lib/filter_migration/lib/__tests__/exists.js

```diff
@@ -0,0 +1,35 @@
+import expect from 'expect.js';
+import { convertExistsFilter } from '../exists';
+
+describe('filter to kuery migration', function () {
+
+  describe('exists filter', function () {
+
+    it('should return a kuery node equivalent to the given filter', function () {
+      const filter = {
+        meta: {
+          type: 'exists',
+          key: 'foo',
+        }
+      };
+      const result = convertExistsFilter(filter);
+
+      expect(result).to.have.property('type', 'function');
+      expect(result).to.have.property('function', 'exists');
+      expect(result.arguments[0].value).to.be('foo');
+    });
+
+    it('should return undefined if the given filter is not of type "exists"', function () {
```

Comment:
If we had a type system passing something different from an "exists" filter as an argument would be a type error. So I would absolutely call this an exceptional scenario. Returning `undefined` just has no intrinsic meaning and can be caused by all kind of accidental mistakes in JS. I would be in favor of clearly signalling the "here is your converted filter" and "I can not convert that". Adding a `supports` function vs throwing an exception essentially comes down to the "ask permission" rather than "forgiveness" pattern. I would be fine with either.